lib/spack/llnl/util/filesystem.py

```diff
@@ -1673,16 +1673,19 @@ def find_first(root: str, files: Union[Iterable[str], str], bfs_depth: int = 2)
     return FindFirstFile(root, *files, bfs_depth=bfs_depth).find()
 
 
-def find(root, files, recursive=True):
+def find(root, files, recursive=True, max_depth: Optional[int] = None):
     """Search for ``files`` starting from the ``root`` directory.
 
     Like GNU/BSD find but written entirely in Python.
 
+    When searching recursively, this behaves as `find` with the `-L`
+    option (follows symlinks).
```

Comment:
very minor: should include `-type f`, and maybe mention that it only considers unique directories, cause `find` does not:

```
tmp.36aYft7vM1 $ find -L . -type f
./b/c/d/e/f/g
./a/symlink/d/e/f/g
tmp.36aYft7vM1 $ realpath $(find -L . -type f)
/tmp/tmp.36aYft7vM1/b/c/d/e/f/g
/tmp/tmp.36aYft7vM1/b/c/d/e/f/g
```